app/src/main/java/com/github/se/icebreakrr/model/profile/ProfilesViewModel.kt

```diff
@@ -98,8 +119,32 @@ open class ProfilesViewModel(
           _profiles.value = profileList
           _filteredProfiles.value = filteredProfiles
           _loading.value = false
+          _isConnected.value = true
+          waitingDone.value = false
+          isWaiting.value = false
+          Log.e(
+              "ConnectionCheck",
+              "Reset flags: waiting=${isWaiting.value}, done=${waitingDone.value}")
         },
-        onFailure = { e -> handleError(e) })
+        onFailure = { e ->
+          Log.e("ConnectionCheck", "Request FAILED")
+          Log.e(
+              "ConnectionCheck",
+              "Current state: waiting=${isWaiting.value}, done=${waitingDone.value}")
+          handleError(e)
+          if (e is com.google.firebase.firestore.FirebaseFirestoreException) {
+            if (!waitingDone.value && !isWaiting.value) {
+              Log.e("ConnectionCheck", "First failure, starting timer")
```

Comment:
Recommendation: Change this to Log.d as it indicates an informational update about the retry mechanism starting, rather than an actual error condition.

Reply:
I am not sure because this is in a onFailure. These errors are not necessarily due to a lack of connection